gfx.cpp

```diff
@@ -9339,12 +9392,12 @@ class GfxInternal
         return kGfxResult_NoError;
     }
 
-    void transitionResource(Buffer &buffer, D3D12_RESOURCE_STATES resource_state)
+    bool transitionResource(Buffer &buffer, D3D12_RESOURCE_STATES resource_state, bool implicit_allowed = false)
```

Comment:
`transitionResource` also now takes and optional parameter which can be used to force disable implicit transition barriers.

Reply:
Could we turn this `bool` into an `enum`?
Say like `kTransitionType_Implicit` vs. `kTransitionType_WhateverIsTheOther`.
I believe that'd make the rest of the code a bit more readable rather than seeing some `transitionResource(..., true);` 🙂

Reply:
I can do that